### ISSUE TITLE
ContentStore.setContent doesn't set content id

### ISSUE THREAD

**Issue by sn42** [paulcwarren/spring-content#57]
Issue is related to spring-content-fs, i didn't check other modules.

## tl;dr
If an entity has a constructor which accepts a single String argument ContentStore's `setContent` method
- doesn't set the entity's field annotated with `@ContentId`
- does create a file which name is equal to the entity's `toString` output (e.g. `package.ClassName@1d9bd4d6`)

[Example Project](https://github.com/sn42/spring-content-conversion-bug-issue)

## Details
The entity's contentId field is only set when a resource is found for that entity. To find the resource the [content store checks](https://github.com/paulcwarren/spring-content/blob/3ac829d3b154d96601fbd853faedeb90c9a81879/spring-content-fs/src/main/java/internal/org/springframework/content/fs/repository/DefaultFilesystemStoreImpl.java#L67) if the conversion service can convert the entity to a string.
If the entity has a constructor which accepts a single string the conversion service can actually convert this entity to a string (entity's `toString` method). This string is then used as the resource location and a non-null resource is returned to `setContent`, which leads to the not-set contentId field and the creation of a file with name equal to the entity's `toString` output.

**Comment by paulcwarren** [comment]
Hi @sn42, thanks for the issue and sample project.  I can confirm this is definitely a bug.

That conversion code is there to allow Developers to add custom location logic.  Sometimes folks want a little bit of control over where content is stored and they might, for example, define an entity with a field that determines a top-level folder and then add a converter that calculates the resource location based on that field and content id).

I was always a little bit dubious about using the conversion service for this purpose.  Semantically, it never felt quite right but at the same time, I didn't want to go inventing something that looked identical.  So perhaps - no we know there is a difference (ignoring String ctors) - one route out of this is to introduce a `PlacementStrategy`.  Let me think about it for a day or so.  Feel free to shout if you have any suggestions too.
